libcloud/test/compute/test_rate_limit.py

```diff
@@ -0,0 +1,39 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import socket
+
+import unittest
+from mock import Mock
+
+from libcloud.common.base import Connection
+
+
+class RateLimitTestCase(unittest.TestCase):
+
+    def test_rate_limit(self):
+        con = Connection()
+        con.connection = Mock()
+        con.connection.request = Mock(side_effect=socket.gaierror)
+
+        try:
+            con.request('/')
+            self.fail('Failed to raise exception')
+        except (Exception, socket.gaierror) as exc:
+            self.assertIsInstance(exc, socket.gaierror,
```

Comment:
All this is testing is that you successfully mocked the request method to raise an exception.  I was thinking about this more last night, what if instead of mocking stuff and calling con.request, just create a ```libcloud.common.base.Response``` object manually with a pre-generated ```httplib.HTTPResponse``` object with the appropriate headers and response code.

Also, I believe there is an ```self.assertRaises``` method that will verify that an exception of a certain type is thrown, rather than manually doing the try/except block.

Something like:
```
from httplib import HTTPResponse
from libcloud.common.base import Response
from libcloud.common.exception import RateLimit

def test_rate_limit(self):
    with self.assertRaises(RateLimit): 
        con = Connection()
        httplib_response = HTTPResponse(figure_out_the_right_args)
        Response(httplib_response, con)
```